src/serialbox/core/SerializerImpl.cpp

```diff
@@ -22,7 +22,7 @@
 #include "serialbox/core/archive/BinaryArchive.h"
 #include "serialbox/core/hash/HashFactory.h"
 #include <boost/algorithm/string.hpp>
-#include <boost/filesystem.hpp>
+#include "Filesystem.h"

```

Comment:
Why not follow the convention with
```
#include "serialbox/core/archive/Filesystem.h"
```
?

Reply:
Because I don't like the convention because you cannot move files without changing all paths... But I will adjust it now and fix the convention later (which probably means never)...

Reply:
Alright but you can write a script which changes all the paths pretty easily with every sane IDE, but point taken. I would just argue that writing
```
#include "serialbox/core/Filesystem.h"
```
is better than this
```
#include "Filesystem.h"
```
because the end user of the library immediately sees this file is part of serialbox while in your (and gridtools for that matter) it's not clear. Imho user experience should take precedence over developer convenience for a library.

Reply:
I think for the cases where you go a level up like

    #include "../core/Filesystem.h"

the absolute path might be a better style (because anyway you introduce a dependency onto something which seems to belong to a different module)

But if you include something from the same directory I don't see why absolute paths are more convenient for the user.

Reply:
I meant for the **end**user (the one who uses the library) ... you can do what you want internally in your library it's just what you expose to the end user that matters. If you really want to do it in a clean way you should split the implementation from the exposed headers (the way LLVM does it, see https://github.com/llvm-mirror/llvm/blob/master/lib/CodeGen/AggressiveAntiDepBreaker.cpp they include private headers directly) -- I did this in the beginning but was forced to change it with the same argument that it was inconvenient for the developer (which I agree in this case).